builder/store/database/mirror.go

```diff
@@ -80,6 +80,35 @@ func (s *MirrorStore) FindByRepoPath(ctx context.Context, repoType types.Reposit
 	return &mirror, nil
 }
 
+func (s *MirrorStore) FindWithMapping(ctx context.Context, repoType types.RepositoryType, namespace, name string, mapping types.Mapping) (*Mirror, error) {
+	var mirror Mirror
+	var err error
+	if mapping == types.CSGHubMapping {
+		return s.FindByRepoPath(ctx, repoType, namespace, name)
+	} else if mapping == types.HFMapping {
+		err = s.db.Operator.Core.NewSelect().
+			Model(&mirror).
+			Relation("Repository").
```

Comment:
The `Relation` method is already called, which automatically joins `repositories` table, so the `Join` method is not needed

Reply:
Fixed